### PR TITLE
v6.6 Changelog and Docs

### PR DESCRIPTION
**Pull request by amyblais**

### CODE REVIEW

**Review by isacikgoz — APPROVED** [review]

**Review by agnivade — APPROVED** [review]

**Review by mkraft — APPROVED** [review]

File: source/install/self-managed-changelog.md [hunk 1]
```diff
@@ -5,12 +5,132 @@
...
+
+**IMPORTANT:** If you upgrade from a release earlier than v6.5, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).
+
+### Highlights
+
+#### Apps Framework 1.0: General Availability
+ - The App Framework allows developers to build integrations with Mattermost that seamlessly work across Mattermost’s desktop and mobile clients, can be coded in any language developers are comfortable with.
+
+#### Actions Restructure
+ - The **Actions** menu was restructured to reduce clutter.
```
Comment (esethna) [inline source/install/self-managed-changelog.md:33]: cc// @aaronrothschild on copy here

File: source/install/self-managed-changelog.md [hunk 2]
```diff
@@ -5,12 +5,132 @@
...
+
+#### Nested Permalink Previews
+ - Added nested previews for permalinks.
```
Comment (esethna) [inline source/install/self-managed-changelog.md:36]: cc// @laneycs on copy here

File: source/install/self-managed-changelog.md [hunk 3]
```diff
@@ -5,12 +5,132 @@
...
+ - Added a post menu item to copy raw text.
+ - Added a loading indicator to the **Threads** global list each time more posts are fetched on infinite scroll.
+ - Added search guidance to the **Threads** global list when no more posts can be loaded. This is only shown if you’ve scrolled to load older posts and reach the end of the list.
+ - Added accessibility support for custom statuses.
+ - Tooltip is now only displayed when text is too long in the announcement banner.
+ - When restricting direct messages to users on the same team, bots are now excluded from that restriction.
+ 
+#### Performance
+ - Improved performance when clearing notifications with Collapsed Reply Threads enabled.
+ - Improved performance of Collapsed Reply Threads when backend is enabled but frontend is disabled.
```
Comment (esethna) [inline source/install/self-managed-changelog.md:58]: Improved performance of Collapsed Reply Threads when `ThreadAutoFollow` is enabled but `CollapsedThreads` is disabled.

File: source/install/self-managed-changelog.md [hunk 4]
```diff
@@ -5,12 +5,132 @@
...
+
+### Go Version
+ - v6.6 is built with Go ``v1.16.7``.
+
+### Open Source Components
+ - Added ``@tippyjs/react``, ``react-popper``, ``react-slidedown`` and ``smooth-scroll-into-view-if-needed`` , and removed ``prettier`` and ``xregexp`` from https://github.com/mattermost/mattermost-webapp.
+
+### Known Issues
+ - [Collapsed Reply Threads](https://docs.mattermost.com/messaging/organizing-conversations.html) is currently in beta. Before enabling the feature, please ensure you are well versed in the [known issues](https://docs.mattermost.com/messaging/organizing-conversations.html#known-issues), particularly relating to database resource requirements and server performance implications. If you cannot easily scale up your database size, or are running the Mattermost application server and database server on the same machine, we recommended waiting to enable Collapsed Reply Threads until it's [promoted to general availability in Q1 2022](https://mattermost.com/blog/collapsed-reply-threads-ga). Learn more about these [performance considerations here](https://support.mattermost.com/hc/en-us/articles/4413183568276).
```
Comment (esethna) [inline source/install/self-managed-changelog.md:119]: ...promoted to general availability in Q2 2022

File: source/install/self-managed-changelog.md [hunk 5]
```diff
@@ -5,12 +5,132 @@
...
+## Release v6.6 - [Feature Release](https://docs.mattermost.com/administration/release-definitions.html#feature-release)
+
+**Release Day: 2022-04-16**
+
+### Compatibility
+ - Updated Safari recommended minimum version to v14.1+.
+
+### Important Upgrade Notes
+ - The Apps Framework protocol for binding/form submissions has changed, by separating the single `call` into separate `submit`, `form`, `refresh` and `lookup` calls. If any users have created their own Apps, they have to be updated to the new system.
+ - The default for ``ThreadAutoFollow`` has been changed to ``true``. This does not affect existing configurations where this value is already set to ``false``.
```
Comment (esethna) [inline source/install/self-managed-changelog.md:23]: We don't really need this to be an important upgrade note since it doesn't affect existing installs

**Review by esethna — APPROVED** [review]
Approving with some suggestions above

File: source/install/self-managed-changelog.md [hunk 6]
```diff
@@ -5,12 +5,132 @@
...
+
+**IMPORTANT:** If you upgrade from a release earlier than v6.5, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).
+
+### Highlights
+
+#### Apps Framework 1.0: General Availability
+ - The App Framework allows developers to build integrations with Mattermost that seamlessly work across Mattermost’s desktop and mobile clients, can be coded in any language developers are comfortable with.
```
Comment (mickmister) [inline source/install/self-managed-changelog.md:30]:
```suggestion
 - The App Framework allows developers to build integrations with Mattermost that seamlessly work across Mattermost’s desktop and mobile clients. Apps can be developed using any programming language, as opposed to plugins which must be developed in Go.
```

**Review by marianunez — APPROVED** [review]

**Review by jasonblais — CHANGES_REQUESTED** [review]
Looking good,
 - for improvements, recommend starting with the benefit/value to the user 
 - recommend adding links to further details/info about a feature or the change. Think from the perspective of someone not familiar with Mattermost and reading the changelog for the first time. They may wonder what "feature flags" or "playbook retrospective metrics" are.

File: source/install/self-managed-changelog.md [hunk 7]
```diff
@@ -5,12 +5,136 @@
...
 
+## Release v6.6 - [Feature Release](https://docs.mattermost.com/administration/release-definitions.html#feature-release)
+
+**Release Day: 2022-04-16**
+
+### Compatibility
+ - Updated Safari recommended minimum version to v14.1+.
+
+### Important Upgrade Notes
+ - The Apps Framework protocol for binding/form submissions has changed, by separating the single `call` into separate `submit`, `form`, `refresh` and `lookup` calls. If any users have created their own Apps, they have to be updated to the new system.
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/install/self-managed-changelog.md:22]: @mickmister has confirmed that a docs update is needed for this update via https://github.com/mattermost/mattermost-webapp/pull/9263#issuecomment-1089362136

File: source/install/self-managed-changelog.md [hunk 8]
```diff
@@ -5,12 +5,136 @@
...
+
+**IMPORTANT:** If you upgrade from a release earlier than v6.5, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).
+
+### Highlights
+
+#### Apps Framework 1.0: General Availability
+ - The App Framework allows developers to build integrations with Mattermost that seamlessly work across Mattermost’s desktop and mobile clients. Apps can be developed using any programming language, as opposed to plugins which must be developed in Go.
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/install/self-managed-changelog.md:29]: https://developers.mattermost.com/integrate/apps/

File: source/install/self-managed-changelog.md [hunk 9]
```diff
@@ -5,12 +5,136 @@
...
+
+#### Actions Restructure
+ - The **Actions** menu was restructured to reduce clutter.
+
+#### Nested Permalink Previews
+ - Added nested previews for permalinks.
+
+#### Playbook Updates
+ - Added retrospective metrics to track up to four key metrics that indicate performance of every run.
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/install/self-managed-changelog.md:38]: @justinegeffen - Can you help provide a link to the docs for this update, please?

File: source/install/self-managed-changelog.md [hunk 10]
```diff
@@ -5,12 +5,136 @@
...
+
+#### Triggers and Actions
+ - Channel admins can now configure certain actions to be executed automatically based on trigger conditions without writing any code.
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/install/self-managed-changelog.md:41]: A PR for Channels is in progress: https://github.com/mattermost/docs/pull/5545
Justine is drafting a Playbook update as well. PR coming soon.

File: source/install/self-managed-changelog.md [hunk 11]
```diff
@@ -5,12 +5,136 @@
...
+ - Updated the plugin registry's ``registerCallButtonAction`` method to allow for displaying custom calls buttons in the channel header.
+ - Added a debugging setting to turn off client-side plugins for the current user.
+ - Added performance metrics related to plugin loading on page load.
+
+#### Administration
+ - The default for ``ThreadAutoFollow`` has been changed to ``true``. This does not affect existing configurations where this value is already set to ``false``.
+ - Improved the license upload flow.
+ - The Start Trial CTA presents a modal exposing the benefits the client gets by starting the trial, encouraging Admins to request a trial license and engage them with the product.
+ - A new field was added to the client configuration to let clients know the database schema version of the server. The applied database migrations have also been added to the **System Console**.
+ - Added a ``Automatically Follow Threads`` configuration setting to the **System Console** to expose the ``threadAutoFollow`` config setting to the User Interface.
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/install/self-managed-changelog.md:77]: Documented via https://github.com/mattermost/docs/pull/5520

File: source/install/self-managed-changelog.md [hunk 12]
```diff
@@ -5,12 +5,136 @@
...
+ - An error is now shown on the email invitation modal if SMTP is not configured but email invitations are ``true``.
+ - Logs from third-party libraries are now included in the default logging configuration.
+ - Added additional performance debugging settings.
+ - The support email field has moved from **Customization** to **Notifications** in the System Console. Also, a support email is now required when configuring email notifications.
+ - The ping endpoint can now receive a device ID, which will report whether the device is able to receive push notifications.
+ - Feature flags are now automatically refreshed when the server undergoes a restart.
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/install/self-managed-changelog.md:83]: https://developers.mattermost.com/contribute/server/feature-flags/

File: source/install/self-managed-changelog.md [hunk 13]
```diff
@@ -5,12 +5,136 @@
...
+ - Changed the Mattermost indigo theme to match the dark theme in code blocks.
+ - Updated in-product links to legacy domain about.mattermost.com.
+ - Added a copy button when hovering over code blocks.
+ - Added a post menu item to copy raw text.
+ - Added a loading indicator to the **Threads** global list each time more posts are fetched on infinite scroll.
+ - Added search guidance to the **Threads** global list when no more posts can be loaded. This is only shown if you’ve scrolled to load older posts and reach the end of the list.
+ - Added accessibility support for custom statuses.
+ - Tooltip is now only displayed when text is too long in the announcement banner.
+ - When restricting direct messages to users on the same team, bots are now excluded from that restriction.
+ - Brazilian Portuguese language support was downgraded to Beta.
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/install/self-managed-changelog.md:57]: Historically, we've documented language support downgrades under Improvements > User Interface (UI) in the changelogs. I appreciate that where this information is included today could be misleading (but we don't necessarily have a better home for this content given the current structure of the the changelog sections).

File: source/install/self-managed-changelog.md [hunk 14]
```diff
@@ -5,12 +5,136 @@
...
+
+#### Playbook Updates
+ - Added retrospective metrics to track up to four key metrics that indicate performance of every run.
+
+#### Triggers and Actions
+ - Channel admins can now configure certain actions to be executed automatically based on trigger conditions without writing any code.
+
+### Improvements
+
+#### User Interface (UI)
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/install/self-managed-changelog.md:45]: @amyblais - I'm happy to help with this ask -- either the initial pass or a follow-up editorial pass. Just let me know how best I can assist.

**Review by jasonblais — APPROVED** [review]
Thanks for the recent updates :+1:

I would encourage adding a link for each highlighted feature (namely triggers and playbook retrospectives) for further details, else LGTM

**Review by larkox — APPROVED** [review]

File: source/install/self-managed-changelog.md [hunk 15]
```diff
@@ -5,12 +5,135 @@
...
+ - Improved the performance of storing users in webapp.
+ - Fixed a small memory leak in the **System Console**.
+
+#### Plugins
+ - Updated the plugin registry's ``registerCallButtonAction`` method to allow for displaying custom calls buttons in the channel header.
+ - Added a debugging setting to turn off client-side plugins for the current user.
+ - Added performance metrics related to plugin loading on page load.
+
+#### Administration
+ - The default for [``ThreadAutoFollow``](https://docs.mattermost.com/configure/configuration-settings.html#automatically-follow-threads) has been changed to ``true``. This does not affect existing configurations where this value is already set to ``false``.
```
Comment (Willyfrog) [inline source/install/self-managed-changelog.md:72]: @esethna should we advise people to turn it on if they have it to off?

**Review by Willyfrog — APPROVED** [review]
Minor comment but looking good 👍🏼

File: source/install/self-managed-changelog.md (hunk 15 above)
Comment (esethna) [reply source/install/self-managed-changelog.md:72]: Good point, I'll add a note

**Pull request merged**